lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -127,6 +127,62 @@ class AwsCompileFunctions {
       const serviceArtifactFileName = this.provider.naming.getServiceArtifactName();
       const functionArtifactFileName = this.provider.naming.getFunctionArtifactName(functionName);
 
+      // create function's iamConfig object
+      functionObject.iamConfig = {
+        principals: new Set(),
+        policyStatements: [],
+        managedPolicies: [],
+      };
```

Comment:
Hi @medikoo . I tried this before but it was giving me multiple errors (~63), which is why I opted for including this code in `functions/index.js` instead. I put the code in `mergeIamTemplates.js` in my latest commit so you can take a look. Maybe there's something I'm missing?

Reply:
> Hi @medikoo . I tried this before but it was giving me multiple errors (~63)

What errors specifically? Is this about failing tests, if so in that case, please list all test files - we need to refactor them to `runServelress` variant.

Technically this upgrade should not require any tests to be written or updated, and just work, but for that we need to have many tests refactored to new approach

Reply:
The errors only occur when the code to create the `functions[].iamConfig` objects are included in `mergeIamTemplates.js` instead of `index.js`. All the errors are from `index.test.js` and they all say this: `TypeError: Cannot read property \'policyStatements\' of undefined` so I think it means the iamConfig object hasn't been created yet. That's why I opted for including this code in 'index.js` instead since I think `mergeIamTemplates.js` hasn't run by the time `index.js` is executed.

Reply:
> All the errors are from index.test.js

This means that we need to address https://github.com/serverless/serverless/issues/8523. After having that, there won't be such error.

Again, let's not get blocked, or have our implementation influenced by failing (for wrong reason) tests. We have very bad tests in many places, and all that fail should be simply refactored (as proposed in above issue)